Assets/MRTK/Tests/PlayModeTests/SpherePointerTests.cs

```diff
@@ -109,6 +123,52 @@ public IEnumerator GrabLayerMasks()
             Assert.True(pointer.IsInteractionEnabled);
         }
 
+        /// <summary>
+        /// Verifies that SpherePointer correctly returns IsNearObject and IsInteractionEnabled
+        /// when 
+        /// </summary>
+        [UnityTest]
+        public IEnumerator GrabLayerMasksWithOverlap()
+        {
+            // Initialize hand
+            var rightHand = new TestHand(Handedness.Right);
+            yield return rightHand.Show(Vector3.zero);
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.OpenSteadyGrabPoint);
+
+            var pointer = rightHand.GetPointer<SpherePointer>();
+            Assert.IsNotNull(pointer, "Expected to find SpherePointer in the hand controller");
+            Vector3 interactionEnabledPos = new Vector3(0.05f, 0, colliderSurfaceZ - pointer.SphereCastRadius);
+
+            Assert.False(pointer.IsNearObject);
+            Assert.False(pointer.IsInteractionEnabled);
+
+            //Initialize overlap cube
+            overlapRect.SetActive(true);
+
+            // Set layer to spatial mesh, which sphere pointer should be ignoring
+            // assumption: layer 31 is the spatial mesh layer
+            cube.SetLayerRecursively(31);
+            yield return null;
+            Assert.False(pointer.IsNearObject);
+            Assert.False(pointer.IsInteractionEnabled);
+
+            // Move hand to object, IsNearObject, IsInteractionEnabled should be true
+            yield return rightHand.MoveTo(interactionEnabledPos);
+            Assert.True(CoreServices.InputSystem.FocusProvider.GetFocusedObject(pointer) == overlapRect, " the overlapping rectangle was not in focus");
```

Comment:
Would be good to verify that isinteractionenabled and isnear object are true here after moving to the interaction enabled pos.